network/stream/syncer.go

```diff
@@ -198,9 +199,24 @@ func RegisterSwarmSyncerClient(streamer *Registry, netStore *storage.NetStore) {
 	})
 }
 
-// NeedData
-func (s *SwarmSyncerClient) NeedData(ctx context.Context, key []byte) (wait func(context.Context) error) {
-	return s.netStore.FetchFunc(ctx, key)
+func (s *SwarmSyncerClient) NeedData(ctx context.Context, key []byte) (loaded bool, wait func(context.Context) error) {
+	start := time.Now()
+
+	fi, loaded, ok := s.netStore.GetOrCreateFetcher(ctx, key, "syncer")
```

Comment:
TODO this logic should be part of netstore.Get no?

Reply:
I don't understand this question. Why do you want this part of `Netstore.Get`? In order to adhere to the interface of the syncer (which this PR is not addressing), we need to know how long to wait for a chunk, hence the returned `Fetcher`, and the select on `<-fi.Delivered` below.

If this is part of `Netstore.Get`, we are just pushing this complexity to the `Netstore.Get`, I don't really see why this is an improvement.

Reply:
Let's leave as it is at the moment, @zelig I've also discussed this at length with @nonsense, there's no benefit of having it here or there. I vote leave as is because the whole syncer part is gonna be heavily simplified anyway and this might as well go into that too